api/src/pcapi/core/providers/titelive_bq_book_search.py

```diff
@@ -73,68 +65,193 @@ def _remove_existing_mediations(self, product: offers_models.Product) -> None:
+                        "Skipping VERSO mediation for EAN %s (UUID %s) due to GCS copy failure.",
+                        product.ean,
+                        bq_product.recto_uuid,
+                    )
+                else:
+                    if current_verso:
+                        db.session.delete(current_verso)
+
+                    verso_mediation = offers_models.ProductMediation(
+                        productId=product.id,
+                        uuid=bq_product.verso_uuid,
+                        imageType=offers_models.ImageType.VERSO,
+                        lastProvider=self.provider,
+                    )
+                    db.session.add(verso_mediation)
 
-    def run_synchronization(self, from_date: datetime.date, to_date: datetime.date) -> None:
-        logger.info("Starting product synchronization from %s to %s.", from_date.isoformat(), to_date.isoformat())
+    def run_synchronization(self, batch_size: int) -> None:
+        logger.info("Starting product synchronization")
 
         total_products_upserted = 0
         ineligible_eans = []
 
-        products_iterator = ProductsToSyncQuery(from_date=from_date, to_date=to_date).execute()
-        for bq_product in products_iterator:
-            reasons = get_ineligibility_reasons(bq_product, bq_product.titre)
+        products_iterator = ProductsToSyncQuery().execute()
 
-            if reasons:
-                if bq_product.gencod in self.product_whitelist_eans:
-                    logger.info(
-                        "Skipping ineligibility for EAN %s (reasons: %s): product is whitelisted.",
-                        bq_product.gencod,
-                        reasons,
-                    )
-                else:
-                    ineligible_eans.append(bq_product.gencod)
-                    logger.info("Rejecting product EAN %s for reason(s): %s.", bq_product.gencod, reasons)
-                    continue
+        for bq_product_batch in get_chunks(products_iterator, batch_size):
+            logger.info("Processing batch of %d products...", len(bq_product_batch))
 
-            product = self.create_or_update(bq_product)
-            try:
-                with transaction():
-                    db.session.add(product)
-                total_products_upserted += 1
-            except sa_exc.IntegrityError as exc:
-                logger.error("Failed to upsert product with EAN %s: %s", product.ean, exc)
+            valid_bq_products_in_batch = []
+
+            for bq_product in bq_product_batch:
+                reasons = get_ineligibility_reasons(bq_product, bq_product.titre)
+
+                if reasons:
+                    if bq_product.ean in self.product_whitelist_eans:
+                        logger.info(
+                            "Skipping ineligibility for EAN %s (reasons: %s): product is whitelisted.",
+                            bq_product.ean,
+                            reasons,
+                        )
```

Comment:
Alors même remarque que précédemment, moi j'ai tendance à faire ça (pour faciliter le filtrage dans le log explorer) : 

```suggestion
                        logger.info(
                            "Skipping ineligibility : product is whitelisted.",
                            extra={
                                "ean": bq_product.ean,
                                "reasons": reasons
                            }
                        )
```